packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js

```diff
@@ -372,8 +447,165 @@ async function askSecurityQuestions(context, parameters) {
   const advancedSettingsAnswer = await inquirer.prompt([advancedSettingsQuestion]);
 
   if (advancedSettingsAnswer.advancedSettings) {
+    authConfig = await askAdditionalAuthQuestions(context, parameters, authConfig, defaultAuthType);
+    if(process.env.SYNC === "true") {
+      resolverConfig = await askResolverConflictQuestion(context, parameters, modelTypes);
+    }
+  }
+
+  return { authConfig, resolverConfig };
+}
+
+async function askResolverConflictQuestion(context, parameters, modelTypes) {
+  let resolverConfig = {};
+
+  if (await context.prompt.confirm('Configure conflict detection?')) {
+    const askConflictResolutionStrategy = async msg => {
+      let conflictResolutionStrategy;
+
+      do {
+        if (conflictResolutionStrategy === 'Learn More') {
+          context.print.info('');
+          context.print.info('DataStore help text');
+          context.print.info('');
+        }
+
+        const conflictResolutionQuestion = {
+          type: 'list',
+          name: 'conflictResolutionStrategy',
+          message: msg,
+          default: 'AUTOMERGE',
+          choices: [
+            {
+              name: 'Auto Merge',
+              value: 'AUTOMERGE',
+            },
+            {
+              name: 'Optimistic Concurrency',
+              value: 'OPTIMISTIC_CONCURRENCY',
+            },
+            {
+              name: 'Custom Lambda',
+              value: 'LAMBDA',
+            },
+            {
+              name: 'Learn More',
+              value: 'Learn More',
+            },
+          ],
+        };
+        ({ conflictResolutionStrategy } = await inquirer.prompt([conflictResolutionQuestion]));
+      } while (conflictResolutionStrategy === 'Learn More');
+
+      let syncConfig = {
+        ConflictHandler: conflictResolutionStrategy,
+        ConflictDetection: 'VERSION',
+      };
+
+      if (conflictResolutionStrategy === 'LAMBDA') {
+        const lambdaFunctionName = await askSyncFunctionQuestion(context);
+        syncConfig.LambdaConflictHandler = {};
+        syncConfig.LambdaConflictHandler.name = lambdaFunctionName;
+      }
+
+      return syncConfig;
+    };
+
+    resolverConfig.project = await askConflictResolutionStrategy('Select the default resolution strategy');
+
+    // Ask for per-model resolver override setting
+
+    if (modelTypes && modelTypes.length > 0) {
```

Comment:
What about having a "Select All" and "Select None" option as well (if possible)?

Reply:
The enquirer plugin supports that out of the box